tee_supplicant/src/tee_supplicant.c

```diff
@@ -120,6 +131,462 @@ static int load_ta(uint32_t num_params, struct tee_param *params)
 	return TEEC_SUCCESS;
 }
 
+static int tee_fs_open(size_t num_params, struct tee_param *params, int flags)
+{
+	struct tee_shm *shm;
+	char *name, path[REE_FS_PATH_MAX] = REE_FS_MP;
+	int fd;
+
+	if (num_params != 3) {
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	if ((params[0].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_VALUE_INPUT ||
+	    (params[1].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_MEMREF_INPUT ||
+	    (params[2].attr & TEE_PARAM_ATTR_TYPE_MASK) !=
+		TEE_PARAM_ATTR_TYPE_VALUE_OUTPUT)
+	{
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	/*TODO do all safety checks */
```

Comment:
It is better to give a comprehensive list of said checks. In two months you will not remember what is actually should be checked here.

Reply:
@dsemenets is already on it, this TODO will be over soon